### PR TITLE
8278568: Consolidate filler objects

### PR DESCRIPTION
**Pull request by rkennke**
In PSPromotionLAB exists some code that fills unused tails of promotion LABs, which is semantically the same as the CollectedHeap::fill* family of methods. They can be consolidated.

We also have ContiguousSpace::allocate_temporary_filler() which does the same, but is unused and can be removed. 

Testing:
 - [x] hotspot_gc
 - [x] tier1
 - [ ] tier2
 - [ ] tier3

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8278568](https://bugs.openjdk.java.net/browse/JDK-8278568): Consolidate filler objects

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Hamlin Li](https://openjdk.java.net/census#mli) (@Hamlin-Li - **Reviewer**) ⚠️ Review applies to f763bbaf5dccaf64cb10a3869609f23b816615a3
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6814/head:pull/6814` \
`$ git checkout pull/6814`

Update a local copy of the PR: \
`$ git checkout pull/6814` \
`$ git pull https://git.openjdk.java.net/jdk pull/6814/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6814`

View PR using the GUI difftool: \
`$ git pr show -t 6814`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6814.diff">https://git.openjdk.java.net/jdk/pull/6814.diff</a>

</details>

### CODE REVIEW

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Review by Hamlin-Li — APPROVED** [review]

File: src/hotspot/share/gc/parallel/psPromotionLAB.cpp [hunk 1]
```diff
@@ -81,20 +75,8 @@ void PSPromotionLAB::flush() {
...
-  assert( (array_length * (HeapWordSize/sizeof(jint))) < (size_t)max_jint, "array too big in PSPromotionLAB");
-  filler_oop->set_length((int)(array_length * (HeapWordSize/sizeof(jint))));
-
-#ifdef ASSERT
-  // Note that we actually DO NOT want to use the aligned header size!
-  HeapWord* elt_words = cast_from_oop<HeapWord*>(filler_oop) + typeArrayOopDesc::header_size(T_INT);
-  Copy::fill_to_words(elt_words, array_length, 0xDEAABABE);
-#endif
+  HeapWord* tlab_end = end() + CollectedHeap::min_dummy_object_size();
+  CollectedHeap::fill_with_object(top(), tlab_end, DEBUG_ONLY(true) NOT_DEBUG(false));
```
Comment (albertnetymk) [inline src/hotspot/share/gc/parallel/psPromotionLAB.cpp:79]: Using `trueInDebug` would be clearer.

File: src/hotspot/share/gc/shared/collectedHeap.hpp [hunk 2]
```diff
@@ -289,7 +289,8 @@ class CollectedHeap : public CHeapObj<mtInternal> {
   }
 
   virtual void fill_with_dummy_object(HeapWord* start, HeapWord* end, bool zap);
-  virtual size_t min_dummy_object_size() const;
+  static size_t min_dummy_object_size();
```
Comment (albertnetymk) [inline src/hotspot/share/gc/shared/collectedHeap.hpp:292]: I wonder if `constexpr` works here.

**Review by albertnetymk — APPROVED** [review]
Just two minor comments/suggestions.

**Comment by rkennke** [comment]
> Just two minor comments/suggestions.

Thanks, good suggestions! I did them both, can you please re-review? Thanks!

**Review by albertnetymk — APPROVED** [review]
Thank you for the revision.

**Review by tschatzl — APPROVED** [review]
Still good.

**Pull request closed without merging**